src/QMCDrivers/WFOpt/QMCCostFunctionBase.cpp

```diff
@@ -397,20 +402,20 @@ bool QMCCostFunctionBase::put(xmlNodePtr q)
     cur = cur->next;
   }
 
-  /// survey all the optimizable objects
-  UniqueOptObjRefs opt_obj_refs;
-  Psi.extractOptimizableObjectRefs(opt_obj_refs);
+  UniqueOptObjRefs opt_obj_refs = extractOptimizableObjects(Psi);
   app_log() << " TrialWaveFunction \"" << Psi.getName() << "\" has " << opt_obj_refs.size()
-            << " optimizable objects:" << std::endl
-            << "  ";
+            << " optimizable objects:" << std::endl;
   for (OptimizableObject& obj : opt_obj_refs)
-    app_log() << " '" << obj.getName() << "'";
-  app_log() << std::endl;
+    app_log() << "   '" << obj.getName() << "'" << (obj.isOptimized() ? " optimized" : " fixed") << std::endl;
 
   //build optimizables from the wavefunction
   OptVariablesForPsi.clear();
-  Psi.checkInVariables(OptVariablesForPsi);
+  for (OptimizableObject& obj : opt_obj_refs)
+    if (obj.isOptimized())
+      obj.checkInVariablesExclusive(OptVariablesForPsi);
   OptVariablesForPsi.resetIndex();
+  app_log() << "Totally " << OptVariablesForPsi.size() << " parameters being optimzied." << std::endl;
```

Comment:
```
<< "This will be radical"
```

Or "In total" for "Totally"

Reply:
Will put "In total" when I finalize #4169